src/nvim/message.c

```diff
@@ -1399,7 +1400,9 @@ void msg_prt_line(char_u *s, int list)
   while (!got_int) {
     if (n_extra > 0) {
       --n_extra;
-      if (c_extra)
+      if (n_extra == 0 && c_final)
```

Comment:
Please add braces to this conditional, we typically want them even when the body of the clauses is 1 line long, for readability.